airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/jsonl_parser.py

```diff
@@ -128,3 +132,96 @@ def _instantiate_accumulator(line: Union[bytes, str]) -> Union[bytes, str]:
             return bytes("", json.detect_encoding(line))
         elif isinstance(line, str):
             return ""
+
+    def parse_records_as_dataframes(
+        self,
+        config: FileBasedStreamConfig,
+        file: RemoteFile,
+        stream_reader: AbstractFileBasedStreamReader,
+        logger: logging.Logger,
+        discovered_schema: Optional[Mapping[str, SchemaType]],
+    ) -> Iterable[pl.DataFrame | pl.LazyFrame]:
+        """Parse records and emit as iterable of data frames.
+
+        Currently this only returns an iterator containing a single data frame. This may
+        be updated in the future to return an iterator with multiple DataFrames.
+        """
+        with stream_reader.open_file(
+            file=file,
+            mode=self.file_read_mode,
+            encoding=None,
+            logger=logger,
+        ) as s3_file:
+
+            def batch_read_lines(
+                file_obj: TextIO,
+                batch_size: int = 500_000,
+            ) -> Iterable[StringIO]:
+                buffer = StringIO()  # Initialize StringIO buffer
+                count = 0
+
+                for line in file_obj:
+                    buffer.write(line)  # Write each line directly to the StringIO buffer
+                    count += 1
+                    if count >= batch_size:
+                        buffer.seek(0)  # Move the pointer to the start of the buffer for reading
+                        yield buffer  # Yield the buffer
+                        buffer = StringIO()  # Reset the buffer for the next batch
+                        count = 0
+
+                if count > 0:  # Yield any remaining lines in the buffer
+                    buffer.seek(0)
+                    yield buffer
+                # buffer: list[str] = []
+                # for line in s3_file:
+                #     buffer.append(line)
+                #     if len(buffer) >= batch_size:
+                #         # Yield the batch as a single string
+                #         yield StringIO("\n".join(buffer))
+                #         buffer = []
+
+            for batch in batch_read_lines(s3_file):
+                df: pl.DataFrame = pl.read_ndjson(
+                    source=batch,
+                    # schema=schema,  # TODO: Add detected schema
+                    infer_schema_length=10,
```

Comment:
Should this be using [this config value](https://github.com/airbytehq/airbyte/pull/44194/files#diff-d6b23b1ee5e90224c8eb3e7fdd7b84041fbaecea90537be165de79fdf1cc2184R98)?